packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Models/Types/ModelTypeProvider.cs

```diff
@@ -121,5 +121,203 @@ private bool PropertyHasSetter(CSharpType type, InputModelProperty prop)
 
             return null;
         }
+
+        protected override CSharpMethod[] BuildConstructors()
+        {
+            List<CSharpMethod> constructors = new List<CSharpMethod>();
+
+            var initializationConstructor = BuildInitializationConstructor();
+            var skipInitializerConstructor = _inputModel.IsUnknownDiscriminatorModel;
+            if (!skipInitializerConstructor)
+            {
+                constructors.Add(initializationConstructor);
+            }
+
+            var serializationConstructor = BuildSerializationConstructor();
+            bool serializationParametersMatchInitialization = serializationConstructor != null &&
+                !serializationConstructor.Signature.Parameters.Any(p => p.Type.IsList) &&
```

Comment:
I think this check is redundant?
We should have the serialization ctor if it is different from the initialization ctor.
For list parameters, their types should be different (IEnumerable<T> vs IList<T>/IReadOnlyList<T>)

Reply:
This condition first checks if there are no list parameters. This is exactly how we check in autorest.csharp - https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProvider.cs#L412

Reply:
well now I think do we really need it? Because when there is a list property, their parameter list will never be the same.

Reply:
Removed `!serializationConstructor.Signature.Parameters.Any(p => p.Type.IsList)` condition.